atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -381,9 +383,9 @@ public void markTableInvolved(TableReference tableRef) {
     @Override
     public Map<byte[], BatchingVisitable<Map.Entry<Cell, byte[]>>> getRowsColumnRange(
             TableReference tableRef, Iterable<byte[]> rows, BatchColumnRangeSelection columnRangeSelection) {
-        return Maps.transformEntries(
-                getRowsColumnRangeIterator(tableRef, rows, columnRangeSelection),
-                (row, iterator) -> BatchingVisitableFromIterable.create(iterator));
+        return KeyedStream.stream(getRowsColumnRangeIterator(tableRef, rows, columnRangeSelection))
+                .map((row, iterator) -> BatchingVisitableFromIterable.create(iterator))
+                .collectTo(() -> new TreeMap<>(UnsignedBytes.lexicographicalComparator()));
```

Comment:
Ah, so the ordering actually comes from `getRowsColumnRangeIterator`. As mentioned elsewhere, is there any way we can be defensive and make sure that we always use a sorted collector? I guess tests would fail if you just used any random collector?

Reply:
Yep, as discussed on the test comments - there are tests that verify that elements are returned in order. 

The interface demands a Map (but not a SortedMap and we can't change that super easily), so the best we can do is maybe something like 

```
    @Override
    public Map<byte[], BatchingVisitable<Map.Entry<Cell, byte[]>>> getRowsColumnRange(
            TableReference tableRef, Iterable<byte[]> rows, BatchColumnRangeSelection columnRangeSelection) {
        SortedMap<byte[], BatchingVisitable<Entry<Cell, byte[]>>> sortedVisitables =
                KeyedStream.stream(getRowsColumnRangeIterator(tableRef, rows, columnRangeSelection))
                        .map((row, iterator) -> BatchingVisitableFromIterable.create(iterator))
                        .collectTo(() -> new TreeMap<>(UnsignedBytes.lexicographicalComparator()));
        return sortedVisitables;
    }
```

though given the tests I think that's unnecessary.

Reply:
Reasonable - I suppose the fact that it is not a SortedMap makes anyone's reliance on the ordering a bit suspicious. And either way, so long as the tests catch this we should catch any regressions.